### ISSUE TITLE
Patch vs. Volume Prediction

### ISSUE THREAD

**Issue by muschellij2** [muschellij2/flexconn#3]
Overall question: how to go from patches to volume?  I have added https://github.com/muschellij2/flexconn/blob/master/R/patches_to_volume.R.  In that I take the "center voxel" to get the probability of lesion.   

There are other ways though.  For example, let's say you have 5 voxels next to each other.  You make patches for each one, then predict, and every voxel in the patch gets a prediction.  Say voxel 1 and voxel 2 are next to each other.  In voxels 1's patch, you get a prediction for voxel 1 and voxel 2.  In voxel 2's patch, you do as well.   Currently, I'm taking the prediction for voxel 1 from the prediction of voxel 1's patch.  I could also take all the predictions when voxel 1 is in a patch, and make a statistic from that (mean).  More complicated, but may be more robust.

Thoughts?

**Comment by skeydan** [comment]
After reading Roy's lmails I think the easiest thing would be to predict on complete images...?
Essentially just translate the python code to R?

In general I've come to think that perhaps we should keep things close to the original python model, and maybe add refinements/evolutions in an upcoming package?

**Comment by muschellij2** [comment]
I think we can have their original prediction, but I think we should have patch-based prediction as an option as it's more intuitive to me that's what the model "expects".  I can add in the original prediction method, though I'm not 100% sure how this would work with 3D.

**Comment by skeydan** [comment]
How about first replicating the way they predict (whole image), to see if that works for us?

(also we would have to remove 
```  L$t2_test = t2_test```

because like that the predict will fail (it needs the same input as during training, apart from batchsize, which will have to be adjusted, but no output)

BTW I'm running a training loop now on 21 atlases, this will take forever as I have to do it on cpu, but I'm saving the model weights, so we can then test the predictions.

**Comment by skeydan** [comment]
... forgot to finish the above, later of course we can test different ways, but I think it would make sense to check their way first

**Comment by muschellij2** [comment]
`L$t2_test = t2_test` only add to the list if `t2` is specified.  If `t2` is `NULL`, then `t2_test = NULL` and then `L$t2_test` will not exist.

**Comment by skeydan** [comment]
oh right, confused it with mask
(actually I hadn't really inspected ```flexconn_model``` before, now I see it's configurable and can have >2 inputs

**Comment by muschellij2** [comment]
I've added `flexconn_predict_with_volume`, which does the https://github.com/muschellij2/flexconn/blob/master/R/flexconn_predict.R#L115 same as `apply_model`.  I've added `remove_small_components` which should mimic https://github.com/skeydan/lesion_segmentation/blob/master/FLEXCONN_Test.py#L205 to get that part done.
